### DISCUSSION TITLE
stacked / exploded raster layers

### DISCUSSION THREAD

**Discussion by UberMouse** [maplibre/maplibre-gl-js#4826] [Ideas]
Our product team has expressed interest in the ability to "z exaggerate / explode" raster layers on the map for analysis. ie render different layers at a different vertical offset so you can see a slice of each layer when looking at a 45* angle instead of only the topmost raster.

Something like this 
![image](https://github.com/user-attachments/assets/1a566780-7f4a-4fed-9c2a-f599ddf286ab)
 which was achieved with vector data using a fill-extrusion layer with a fill-extrusion-base and fill-extrusion-height set to generate the vertical offset between layers.

That functionality is effectively what they are after I believe, but it's only applicable to vector sources not raster sources which is what they want.

Is this possible through some other mechanism that I don't know about? If not, is this something maplibre has any interest in supporting?

**Comment by HarelM** [comment]
You might be able to achieve this use a custom layer, as you have reference to the gl context.

**Reply by UberMouse** [reply]
Yes we are well aware of custom layers, we have a custom COPC (Point Cloud) renderer. We were hoping to not have to completely recreate rendering raster tiles from scratch (which is I think how it would have to work with a custom layer?) considering we want it to work exactly as it currently does, but with a z-offset.

We could potentially contribute changes to ML if it's something of interest.

**Reply by HarelM** [reply]
I'm not sure if this is something people would need, so I think a plugin might be a better approach ATM.
